src/common/Makefile.am

```diff
@@ -69,14 +64,32 @@ libcommon_internal_la_SOURCES = \
 	common/addr_parsing.c \
 	common/hobject.cc \
 	common/bloom_filter.cc \
-	common/linux_version.c \
 	common/module.c \
 	common/Readahead.cc \
 	common/Cycles.cc \
 	common/ContextCompletion.cc
 
+if LINUX
+libcommon_internal_la_SOURCES += \
+	common/xattr.c \
+	common/ipaddr.cc \
+	common/ceph_json.cc \
+	common/util.cc \
+	common/pick_address.cc \
+	common/linux_version.c 
+endif
```

Comment:
@rohanmars the source files above compile on osx and very likely on freebsd. yeah, other than `linux_version.c` does not make sense on non-linux platforms. but so far, we'd like to compile the daemons on osx/freebsd, and the store backends are using `get_linux_version()` do perform some checks. so, we can not just drop it.